packages/tizen_notification/lib/src/types.dart

```diff
@@ -2,102 +2,189 @@
+  NotificationIcons({
+    this.icon,
+    this.indicatorIcon,
+    this.lockIcon,
   });
 
-  /// The path of icon.
-  final String? iconPath;
+  /// The path to the icon file.
+  final String? icon;
 
-  /// The path of indicator icon.
-  final String? indicatorPath;
+  /// The path to the indicator icon file.
+  final String? indicatorIcon;
 
-  /// The path of lock screen icon.
-  final String? lockPath;
+  /// The path to the lock screen icon file.
+  final String? lockIcon;
 
-  /// Returns [NotificationImage] member fields in a map format.
-  Map<String, dynamic> toMap() => <String, dynamic>{
-        'icon': iconPath,
-        'iconForIndicator': indicatorPath,
-        'iconForLock': lockPath,
+  /// Converts to a map.
+  Map<String, Object?> toMap() => <String, Object?>{
+        'icon': icon,
+        'iconForIndicator': indicatorIcon,
+        'iconForLock': lockIcon,
       };
 }
 
+/// The type of sound.
+enum SoundType {
+  /// No sound.
+  none,
+
+  /// Default sound.
+  builtIn,
+
+  /// User sound data.
+  userData,
+}
+
+/// The sound to play when the notification is presented.
 class NotificationSound {
-  /// [NotificationSound] specifies sound options for notifications.
-  ///
-  /// [path] refers to a file path to custom sound data played on notification,
-  /// this value is ignored When [type] is not [SoundType.userData].
-  NotificationSound({
-    required this.type,
-    this.path,
-  });
+  /// Creates a [NotificationSound] with the given [type] and [path].
+  NotificationSound({required this.type, this.path});
 
+  /// The type of sound.
   final SoundType type;
+
+  /// The path to the user sound data file.
+  ///
+  /// Only applicable if the [type] is [SoundType.userData].
   String? path;
 
-  /// Returns [NotificationSound] member fields in a map format.
-  Map<String, dynamic> toMap() => <String, dynamic>{
+  /// Converts to a map.
+  Map<String, Object?> toMap() => <String, Object?>{
         'type': type.name,
         'path': path,
       };
 }
 
+/// The type of vibration.
+enum VibrationType {
+  /// No vibration.
+  none,
+
+  /// Default vibration pattern.
+  builtIn,
+
+  /// User vibration data.
+  userData,
+}
+
+/// The notification vibration options.
 class NotificationVibration {
-  /// [NotificationVibration] specifies vibration options for notifications.
-  NotificationVibration({
-    required this.type,
-    this.path,
-  });
+  /// Creates a [NotificationVibration] with the given [type] and [path].
+  NotificationVibration({required this.type, this.path});
 
+  /// The type of vibration.
   final VibrationType type;
+
+  /// The path to the user vibration data file.
+  ///
+  /// Only applicable if the [type] is [VibrationType.userData].
   String? path;
 
-  /// Returns [NotificationVibration] member fields in a map format.
-  Map<String, dynamic> toMap() => <String, dynamic>{
+  /// Converts to a map.
+  Map<String, Object?> toMap() => <String, Object?>{
         'type': type.name,
```

Comment:
If you use the actual values in `notification_type.h` ​​corresponding to each enumerable value instead of `name`, you don't have to convert name string to enum in native imple.

Reply:
I personally think it's not a good practice to use raw native constants in Dart code as I mentioned above.

Reply:
Ok, I agree with you.